src/polyglot_ffi/generators/c_stubs_gen.py

```
@@ -93,36 +106,56 @@ def _generate_function_stub(self, func: IRFunction) -> list:
-        # Declare local variables
-        num_locals = len(func.params) + 1
-        local_vars = ", ".join(f"ml_{p.name}" for p in func.params) + ", ml_result"
-        lines.append(f"    CAMLlocal{num_locals}({local_vars});")
+        # Declare local variables (only for non-unit parameters)
+        # Note: we still need ml_result
+        if non_unit_params:
+            num_locals = len(non_unit_params) + 1
+            local_vars = ", ".join(f"ml_{p.name}" for p in non_unit_params) + ", ml_result"
+            lines.append(f"    CAMLlocal{num_locals}({local_vars});")
+        else:
+            lines.append(f"    CAMLlocal1(ml_result);")
```

Ruff (0.14.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)